web/src/components/RawDataEditor/RawDataEditor.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useRef } from 'react';
+import { Button, Drawer, PageHeader, Tooltip, notification } from 'antd';
+import { LinkOutlined } from '@ant-design/icons';
+import CodeMirror from '@uiw/react-codemirror';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { useIntl } from 'umi';
+
+type Props = {
+  visible: boolean,
```

Comment:
both `visible` or `readonly` is optional, the default value is `false`.

Reply:
no, both `visible` and  `readonly` are required.

Reply:
why they are required?

Reply:
`visible`:  If you set visible to optional, then the user can use this component without passing this variable, then the component will not be displayed.
`readonly`:  Want the user to accurately call the state of the component